apps/src/templates/progress/progressHelpers.js

```diff
@@ -194,3 +206,87 @@ export const processedLevel = level => {
       level.sublevels && level.sublevels.map(level => processedLevel(level))
   };
 };
+
+const getLevelResult = serverProgress => {
+  if (serverProgress.status === LevelStatus.locked) {
+    return TestResults.LOCKED_RESULT;
+  }
+  if (serverProgress.readonly_answers) {
+    return TestResults.READONLY_SUBMISSION_RESULT;
+  }
+  if (serverProgress.submitted) {
+    return TestResults.SUBMITTED_RESULT;
+  }
+
+  return serverProgress.result || resultFromStatus(serverProgress.status);
```

Comment:
see `progressTypes.studentLevelProgressType` for description of these properties and pointers to their "enum" definitions. 

as you know, we have a lot of technical debt here, and throughout the refactor work it's been necessary to strike a balance between auditing the old code for potential tech debt cleanup vs. preserving legacy logic as-is. cleaning up the relationship between `result` and `status` is going to require a deep dive, so for anything pertaining to that, i opted for preserving logic as-is where possible. 

previously, when processing section progress data in `sectionProgressLoader`, we would pass all the result values returned from the server through [progressRedux.getLevelResult](https://github.com/code-dot-org/code-dot-org/blob/3adbeb6b5ec6227c9d5f114d4ee93cbcd6cc683e/apps/src/code-studio/progressRedux.js#L286), presumably so that the first three conditional blocks could potentially override a server result value. i never dug into why (or even if) that part of this "algorithm" is necessary.

`getLevelResult` was moved into this module as part of the `progressRedux` refactor that got punted, and was expanded to add `|| resultFromStatus` to handle part of said refactor. leaving this function in two places was an oversight when i extracted this work into its own PR. i will update `progressRedux` to use this version.